### ISSUE TITLE
`if-then-panic` suggestion discards comments

### ISSUE THREAD

**Issue by mikerite** [rust-lang/rust-clippy#7730]
I tried this code:

```rust
#![warn(clippy::if_then_panic)]

fn main() {
    if false { 
        // comment
        panic!("ABC"); 
    };
}
```

I expected to see this happen: The suggestion would preserve the comment somehow.

Instead, this happened: The suggestion discarded the comment.
```
warning: only a `panic!` in `if`-then statement
 --> src/main.rs:5:5
  |
5 | /     if false { 
6 | |         // comment
7 | |         panic!("ABC"); 
8 | |     };
  | |_____^ help: try: `assert!(!false, "ABC");`
  |
```

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.57.0-nightly (fdf65053e 2021-09-07)
binary: rustc
commit-hash: fdf65053e99e8966f9bd83b5a8491326cb33d638
commit-date: 2021-09-07
host: x86_64-unknown-linux-gnu
release: 1.57.0-nightly
LLVM version: 13.0.0
```
@rustbot label +L-suggestion
